src/addressResolvers/ens.ts

```diff
@@ -38,3 +38,12 @@ export async function lookupAddresses(addresses: Address[]): Promise<Record<Addr
     return {};
   }
 }
+
+export async function resolveName(handle: string): Promise<string | undefined> {
+  try {
+    const addressResolved = await provider(NETWORK).resolveName(handle);
```

Comment:
Better to move `provider(NETWORK)` outside this function so it can be reused, - NOT too important